### ISSUE TITLE
`cy.intercept` on file upload endpoints seems to be reading the file stream, which can result in server-side throwing error

### ISSUE THREAD

**Issue by JohnnyFun** [cypress-io/cypress#9534]
This is easy to work around, so don't stress about it too much. Just wanted to report in case it helps.

### Current behavior
If I `cy.intercept('POST', /profile\/picture/i).as('upload')` to watch an endpoint that consumes a file upload, my .net core 3 server errors with "System.IO.IOException: Unexpected end of Stream, the content may have already been read by another component."

### Desired behavior
Maybe cypress just needs to set the stream position back to 0?

### Versions
- Cypress@6.0.1
- Chrome 87 (the one cypress opens)
- Win10

**Comment by JohnnyFun** [comment]
A co-worker pointed out that my loose hypothesis about setting position back to 0 doesn't make sense since it'd probably be getting forwarded via http to our server. 

I'll look more at this related ticket and see if I can dig up more info: https://github.com/dotnet/aspnetcore/issues/18087

**Comment by JohnnyFun** [comment]
Oh and if it's any help, this test worked with the old `cy.route`.

**Comment by emilong** [comment]
@JohnnyFun I might be having a similar issue - could you say more about your workaround? (Mine is to use `cy.route()` as well)

**Comment by JohnnyFun** [comment]
I ended up just using ui loading indicators to know when the network request is completed. So something like this:

```
cy.get('.loading-thing').should('exist')
uploadFile()
cy.get('.loading-thing').should('not.exist')
```

**Comment by jennifer-shehane** [comment]
Right now there doesn't seem to be enough information to reproduce the problem on our end. We'll have to close this issue until we can reproduce it. Please comment in this issue with a reproducible example and we will consider reopening the issue.

Also many of the initial intercept bugs have been fixed in our more recent releases, so please update to the latest version. This issue may have been fixed.

**Comment by emilong** [comment]
FYI The issue I had here was fixed in 6.2.1 🎉
